pkg/client/connector/install_actions.go

```diff
@@ -362,6 +362,11 @@ func (ata *addTrafficAgentAction) Do(obj kates.Object) error {
 		return install.ObjErrorf(obj, "unable to find app container %q in", ata.containerName)
 	}
 
+	// Under some odd circumstances, the agent volume can be left over after an uninstall.
+	// Drop it if we get here and it's present, since it'll cause errors.
+	// We ignore the error from this since we don't care if the volume isn't already present
+	_ = ata.dropAgentAnnotationVolume(obj, tplSpec)
```

Comment:
I think these odd circumstances are only in tests and using the smart agent bc the test version is something like this: `v2.0.0-gotest.53272` which means system a gives us the *pinned* agent image: docker.io/datawire/ambassador-telepresence-agent:1.2.0 , and since our current version of telepresence is not backwards compatible with that version, it's not surprising that weirdness ensues.

So I'm wondering if we should change the GenGo to create a v2.y.z that is closer to reality so we don't run into this bug.. but I think we are fine w/ leaving this function in, just wanted to explain why I think it was happening since it just happened to me.